### PR TITLE
101 decoupled action spaces

### PR DESCRIPTION
**Pull request by AhmadAmine998**

### CODE REVIEW

File: gym/f110_gym/envs/action.py [hunk 1]
```diff
@@ -19,73 +19,134 @@ def from_string(action: str):
...
 
-class SpeedAction(CarAction):
+class SteeringAngleAction(SteerAction):
     def __init__(self, params: Dict) -> None:
         super().__init__()
-        self._action_type = "speed"
-
-        self.steering_low, self.steering_high = params["s_min"], params["s_max"]
-        self.velocity_low, self.velocity_high = params["v_min"], params["v_max"]
+        self._type = "speed"
```
Comment (hzheng40) [inline gym/f110_gym/envs/action.py:92]: typo? also should add steering so it's specified

File: gym/f110_gym/envs/action.py [hunk 2]
```diff
@@ -19,73 +19,134 @@ def from_string(action: str):
...
-            params["v_max"],
-            params["v_min"],
         )
-        return accl, sv
+        return sv
+    
+class SteeringSpeedAction(SteerAction):
+    def __init__(self, params: Dict) -> None:
+        super().__init__()
+        self._type = "speed"
```
Comment (hzheng40) [inline gym/f110_gym/envs/action.py:108]: add steering so it's specified

File: gym/f110_gym/envs/action.py [hunk 3]
```diff
@@ -19,73 +19,134 @@ def from_string(action: str):
...
+    ) -> float: 
+        return action
+
+class SteerActionEnum(Enum):
+    Angle = 1
+    Speed = 2
+
+    @staticmethod
+    def from_string(action: str):
+        if action == "angle":
```
Comment (hzheng40) [inline gym/f110_gym/envs/action.py:122]: again, should specifiy steering_angle and steering_speed, just to be less confusing

**Review by hzheng40 — CHANGES_REQUESTED** [review]

**Comment by hzheng40** [comment]
@AhmadAmine998 I pulled in the updated tests and it's failing right now. You'll probably have to update the tests to use the decoupled action spaces.
Also, should maybe consider handling when only one action type is set, (throw a warning and setting the other to default).

**Comment by AhmadAmine998** [comment]
> @AhmadAmine998 I pulled in the updated tests and it's failing right now. You'll probably have to update the tests to use the decoupled action spaces. Also, should maybe consider handling when only one action type is set, (throw a warning and setting the other to default).

I updated the initialization of the action space to be compatible with specifying only one control input as a string. The old tests passed and I now updated the tests to use the decoupled action specification. This way it is compatible with legacy code while still having updated tests

**Review by hzheng40 — APPROVED** [review]
Looks good.

**Pull request merged**